diff --git a/ldlidar_component/CMakeLists.txt b/ldlidar_component/CMakeLists.txt
index 763bde8..1bea53f 100644
--- a/ldlidar_component/CMakeLists.txt
+++ b/ldlidar_component/CMakeLists.txt
@@ -199,7 +199,7 @@ rclcpp_components_register_nodes(${PROJECT_NAME} "ldlidar::LdLidarComponent")
 set(node_plugins "${node_plugins}ldlidar::LdLidarComponent;$<TARGET_FILE:${PROJECT_NAME}>\n")
 
 # Install components
-install(TARGETS ${PROJECT_NAME}
+install(TARGETS ${PROJECT_NAME} ldlidar
   ARCHIVE DESTINATION lib
   LIBRARY DESTINATION lib
   RUNTIME DESTINATION lib/${PROJECT_NAME}
diff --git a/ldlidar_component/component/src/ldlidar_component.cpp b/ldlidar_component/component/src/ldlidar_component.cpp
index 89865a6..44a1f8a 100644
--- a/ldlidar_component/component/src/ldlidar_component.cpp
+++ b/ldlidar_component/component/src/ldlidar_component.cpp
@@ -510,7 +510,7 @@ void LdLidarComponent::lidarThreadFunc()
     }
     // <---- Interruption check
 
-    int nSub = count_subscribers(_scanTopic);
+    int nSub = _scanPub->get_subscription_count();
     if (nSub > 0) {
       _publishing = true;
       switch (_lidar->GetLaserScanData(laser_scan_points, _readTimeOut_msec)) {
